rllib/examples/connectors/connector_v2_nested_action_spaces.py

```diff
@@ -0,0 +1,94 @@
+from gymnasium.spaces import Dict, Tuple, Box, Discrete, MultiDiscrete
+
+from ray.tune.registry import register_env
+from ray.rllib.connectors.env_to_module import (
+    AddObservationsFromEpisodesToBatch,
+    FlattenObservations,
+    WriteObservationsToEpisodes,
+)
+from ray.rllib.examples.env.multi_agent import MultiAgentNestedSpaceRepeatAfterMeEnv
+from ray.rllib.examples.env.nested_space_repeat_after_me_env import (
+    NestedSpaceRepeatAfterMeEnv,
+)
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+    run_rllib_example_script_experiment,
+)
+from ray.tune.registry import get_trainable_cls
+
+
+# Read in common example script command line arguments.
+parser = add_rllib_example_script_args(default_timesteps=200000, default_reward=-500.0)
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    # Define env-to-module-connector pipeline for the new stack.
+    def _env_to_module_pipeline(env):
+        return [
+            AddObservationsFromEpisodesToBatch(),
+            FlattenObservations(multi_agent=args.num_agents > 0),
+            WriteObservationsToEpisodes(),
+        ]
+
+    # Register our environment with tune.
+    if args.num_agents > 0:
+        register_env(
+            "env",
+            lambda c: MultiAgentNestedSpaceRepeatAfterMeEnv(
+                config=dict(c, **{"num_agents": args.num_agents})
+            ),
+        )
+    else:
+        register_env("env", lambda c: NestedSpaceRepeatAfterMeEnv(c))
+
+    # Define the AlgorithmConfig used.
+    base_config = (
+        get_trainable_cls(args.algo)
+        .get_default_config()
+        .environment(
+            "env",
+            env_config={
+                "space": Dict(
+                    {
+                        "a": Tuple(
+                            [Dict({"d": Box(-15.0, 3.0, ()), "e": Discrete(3)})]
+                        ),
+                        "b": Box(-10.0, 10.0, (2,)),
+                        "c": MultiDiscrete([3, 3]),
+                        "d": Discrete(2),
+                    }
+                ),
+                "episode_len": 100,
+            },
+        )
+        .rollouts(env_to_module_connector=_env_to_module_pipeline)
```

Comment:
No `env_runner_cls`  -> we are still in the old stack or do we use them now by default?

Reply:
Great question! I changed the utility function used in all example scripts now to automatically add some more config settings (hence: `base_config` instead of `config`) based on the provided command line args.